torch/csrc/dynamo/guards.cpp

```diff
@@ -1939,6 +2006,11 @@ class RootGuardManager : public GuardManager {
     _relational_guard_resetters.emplace_back(std::move(relational_guard));
   }
 
+  void add_relational_guard_resetter_without_move(
+      std::shared_ptr<RelationalGuard> relational_guard) {
+    _relational_guard_resetters.emplace_back(relational_guard);
```

Comment:
```suggestion
    _relational_guard_resetters.emplace_back(std::move(relational_guard));
```

Reply:
Surprised this isn't covered by clang-tidy. A lot of things should probably moved by value around here.